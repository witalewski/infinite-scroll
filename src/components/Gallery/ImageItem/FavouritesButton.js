import React from 'react';
import styled from '@emotion/styled';

const FavouritesButtonStyled = styled.button`
  position: absolute;

  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  width: 120px;
  height: 120px;
  padding-top: 12px;
  font-size: 64px;

  background: rgba(255, 255, 255, 0.8);
  border: none;

  transition: opacity 0.5s;

  &.btn--hidden {
    opacity: 0;
  }
`;

export const FavouritesButton = ({ visible, isFavourite, onButtonClick }) => (
  <FavouritesButtonStyled
    className={`btn btn-light ${!visible && 'btn--hidden'}`}
    type="button"

  >
    {isFavourite ? '💔' : '❤️'}
  </FavouritesButtonStyled>
);
