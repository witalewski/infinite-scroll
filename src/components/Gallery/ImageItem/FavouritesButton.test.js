import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { FavouritesButton } from './FavouritesButton';

describe('FavouritesButton', () => {
  it('matches snapshot for non-favourite', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FavouritesButton isFavourite={false} visible={true} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot for favourite', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FavouritesButton isFavourite={true} visible={true} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
