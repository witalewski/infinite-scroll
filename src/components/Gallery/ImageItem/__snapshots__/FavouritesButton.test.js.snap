// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesButton matches snapshot for favourite 1`] = `
<ForwardRef(render)
  className="btn btn-light false"
  type="button"
>
  💔
</ForwardRef(render)>
`;

exports[`FavouritesButton matches snapshot for non-favourite 1`] = `
<ForwardRef(render)
  className="btn btn-light false"
  type="button"
>
  ❤️
</ForwardRef(render)>
`;
